### PR TITLE
Add github action for commercial E2E tests

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
Add a workflow to run our cypress e2e tests when changes are made to the commercial code.

It pulls DCR and this repositiory, builds the commercial bundle only from frontend and runs all tests against DCR. Avoiding the hassle of running frontend in CI.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
Continuous automated e2e testing helps catch issues with new code changes before they're deployed.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: cypress/fixtures/breakpoints.ts [hunk 1]
```diff
@@ -7,10 +7,10 @@ type BreakpointSizes = {
 };
 
 const breakpointsToTest: Array<keyof typeof breakpoints> = [
-	'mobile',
+	// 'mobile',
 	'tablet',
-	'desktop',
-	'wide',
+	// 'desktop',
+	// 'wide',
```
Comment (SiAdcock) [inline cypress/fixtures/breakpoints.ts:13]: Do we want to reinstate these breakpoints?

File: cypress/lib/util.ts [hunk 2]
```diff
@@ -8,26 +8,33 @@ import type {UserFeaturesResponse} from '../../static/src/javascripts/types/memb
...
 			}
 		}
 	}
+
+	if (adtest) {
+		const builder = new URL(url);
+		builder.searchParams.append('adtest', adtest);
+		url = builder.toString();
+	}
+	return url;
```
Comment (SiAdcock) [inline cypress/lib/util.ts:37]: Much nicer 👏

File: cypress/fixtures/breakpoints.ts (hunk 1 above)
Comment (Jakeii) [reply cypress/fixtures/breakpoints.ts:13]: Good catch!

**Review by arelra — APPROVED** [review]
Fantastic work @Jakeii and team in getting to this point.

A big milestone for commercial dev 🎉

**Pull request merged**